dvc/repo/__init__.py

```diff
@@ -390,16 +402,8 @@ def stages(self, from_directory=None, check_dag=True):
                         outs.append(out.fspath + out.sep)
                 stages.append(stage)
 
-            def filter_dirs(dname, root=root):
-                path = os.path.join(root, dname)
-                if path in (self.dvc_dir, self.scm.dir):
-                    return False
-                for out in outs:
-                    if path == os.path.normpath(out) or path.startswith(out):
-                        return False
-                return True
-
-            dirs[:] = list(filter(filter_dirs, dirs))
+            out_dir_filter = self.make_out_dir_filter(outs, root)
+            dirs[:] = list(filter(out_dir_filter, dirs))
```

Comment:
Btw, we've talked about this being actually dvcignore-like logic and that maybe it would make sense to unify it. But, I guess two parts of dvcignore logic (ignore specific data and ignore dvc-files) got into conflict here?

Reply:
Do not look for stage files inside outs is not a dvcignore logic I guess.

Reply:
On the other hand this (ignoring) is functionality that should be taken by dvcignore... How about creating class DvcIgnoreFromOuts and adding it to dvcignore here?

Reply:
This has nothing to do with dvcignore, and we only don't look for stage files there, we look for everything else.